### DISCUSSION TITLE
Creating pages from data files (json, yaml, etc) - possible?

### DISCUSSION THREAD

**Discussion by taviks** [lumeland/lume#40] [Q&A] [answered]
Greetings,

Does Lume support creating pages from data files, a la https://www.11ty.dev/docs/pages-from-data/?

**Comment by oscarotero** [comment, marked as answer]
Yes, you have to create a page in javascript or typescript exporting a generator. You have an example with pagination here: https://lumeland.github.io/creating-pages/pagination/

Basically, every value returned by a generator will be converted to a page. The 11ty example would be something like this:

```js
// possum-pages.tmpl.js

export default function *({possums}) {
    for (const possum of possums) {
        yield {
            url: `possums/${possum.name}/`,
            content: `${possum.name} is ${possum.age} years old`
        }
    }
}
```

**Comment by taviks** [comment]
Excellent! Now I see that in the docs 🤦 

Thanks for clarifying!

**Comment by shadowtime2000** [comment]
@oscarotero Maybe convert this issue to a discussion in section "Q&A"?

**Comment by oscarotero** [comment]
@shadowtime2000 That's a good idea. Can we move/convert this issue to a discussion? Or we need to create a new discusion from scratch?

**Comment by shadowtime2000** [comment]
@oscarotero I think you will be able to convert this issue to a discussion, just click "Convert to discussion" where it says pin issue and transfer issue.

**Reply by oscarotero** [reply]
Awesome, thanks!
